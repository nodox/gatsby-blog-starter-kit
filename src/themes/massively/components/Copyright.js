import React from 'react';


export default class Copyright extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div id="copyright">
        <ul>
          <li>{this.props.data.owner}</li>
          <li>Design: <a href="https://html5up.net">HTML5 UP</a></li>
          <li>Powered by: <a href="https://www.gatsbyjs.org/">GatsbyJS</a></li>
        </ul>
      </div>
    );
  }
}
