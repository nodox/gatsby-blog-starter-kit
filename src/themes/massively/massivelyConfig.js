module.exports = {
    initialLinks: {
      paths: [
        {
          path: '/',
          title: 'Home',
        },
        {
          path: '/about',
          title: 'About'
        },
        {
          path: '/speaking',
          title: 'Speaking'
        },
        {
          path: '/contact',
          title: 'Contact'
        },
      ],
      socialLinks: [
        {
          path: 'https://www.twitter.com/stevennatera',
          icon: 'fa-twitter',
          name: 'Twitter',
        },
        {
          path: 'https://www.linkedin.com/in/snatera',
          icon: 'fa-linkedin',
          name: 'LinkedIn',
        },
        {
          path: 'https://www.github.com/nodox',
          icon: 'fa-github',
          name: 'Github',
        }
      ]
    },
    copyright: {
      owner: 'Â© Steven Natera',
    },
    introText: {
      '/': 'Hi! I\'m Steven Natera. I write this blog, teach software engineering workshops, and explore emerging technologies.',
      '/about': 'Hi! I\'m Steven Natera. I write this blog, teach software engineering workshops, and explore emerging technologies.',
      '/speaking': 'I deliver talks on NodeJS and I teach software engineering workshops.',
      '/contact': 'I\'m pretty easy to get hold of, here\'s how I use different channels to communicate with people and how best to contact me'
    }
}
