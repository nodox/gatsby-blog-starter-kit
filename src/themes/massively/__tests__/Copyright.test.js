import React from 'react';
import renderer from 'react-test-renderer';
import Copyright from '../components/Copyright';
import * as config from '../massivelyConfig';

describe('<Copyright />', () => {
  const component = mount(
    <Copyright data={config.copyright} />
  );

  const props = component.instance().props;

  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('should have the right props', () => {
    expect(props).toHaveProperty('data');
    expect(props).toHaveProperty('data.owner');
  });

});
