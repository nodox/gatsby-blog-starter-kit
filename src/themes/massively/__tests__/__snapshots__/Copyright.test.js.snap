// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Copyright /> should render correctly 1`] = `
<Copyright
  data={
    Object {
      "owner": "© Steven Natera",
    }
  }
>
  <div
    id="copyright"
  >
    <ul>
      <li>
        © Steven Natera
      </li>
      <li>
        Design: 
        <a
          href="https://html5up.net"
        >
          HTML5 UP
        </a>
      </li>
      <li>
        Powered by: 
        <a
          href="https://www.gatsbyjs.org/"
        >
          GatsbyJS
        </a>
      </li>
    </ul>
  </div>
</Copyright>
`;
